src/test/java/com/google/api/codegen/util/ProtoParserTest.java

```diff
@@ -132,33 +144,34 @@ public void testIsHttpGetMethod() {
   @Test
   public void testGetServiceAddress() {
     String defaultHost = protoParser.getServiceAddress(libraryService);
-    assertThat(defaultHost).isEqualTo("library-example.googleapis.com");
+    assertThat(defaultHost).isEqualTo("library-example.googleapis.com:1234");
   }
 
   @Test
   public void testGetRequiredFields() {
     Method publishSeriesMethod = libraryService.lookupMethod("PublishSeries");
     List<String> requiredFields = protoParser.getRequiredFields(publishSeriesMethod);
     Collections.sort(requiredFields);
-    assertThat(requiredFields.size()).isEqualTo(2);
+    assertThat(requiredFields.size()).isEqualTo(3);
     assertThat(requiredFields.get(0)).isEqualTo("books");
-    assertThat(requiredFields.get(1)).isEqualTo("shelf");
+    assertThat(requiredFields.get(1)).isEqualTo("series_uuid");
+    assertThat(requiredFields.get(2)).isEqualTo("shelf");
```

Comment:
I think you can leave `requiredFields` unsorted then `assertThat(requiredFields).containsExactly("books", "series_uuid", "shelf");`

Reply:
neat